### PR TITLE
Differentiate line and block comments

### PR DESCRIPTION
**Pull request by Aerijo**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

Tree-sitter currently treats all comments as `comment.block`. However, the distinction between block and line can be useful

For example, I'm trying to write a script that auto-comments newlines when created inside a line comment. I don't want it to do this inside a block comment, but it's difficult to tell which it is using the provided scopes. E.g., the desired behaviour is
```
// some |text

// some
// |text
```

but in an actual block comment, we would get
```
/*
some |text
*/

/*
some
// |text
*/
```

### Alternate Designs

The Tree-sitter grammar could differentiate comments.

### Benefits

More precise scoping

### Possible Drawbacks

Regular expressions can hurt performance. Hopefully, it can optimise this one, as it's only a constant two character check each time. It'd still feel better if there was a rule to match an exact start string (so just `//`).

### Applicable Issues

NA

### CODE REVIEW

File: grammars/tree-sitter-javascript.cson [hunk 1]
```diff
@@ -132,7 +132,13 @@ scopes:
   'null': 'constant.language.null'
   'true': 'constant.language.boolean.true'
   'false': 'constant.language.boolean.false'
-  'comment': 'comment.block'
+  'comment': [
+    {
+      match: "^//",
```
Comment (thomasjo) [inline grammars/tree-sitter-javascript.cson]: Line comments can appear anywhere on a line; this only handles cases where it's at the beginning of a line.
We need tests that explicitly handles various line comment scenarios.

Comment (Aerijo) [reply grammars/tree-sitter-javascript.cson]: @thomasjo The regex here refers to the start of the comment node, which should always be `//` if it's a simple line comment. It will catch comments at the end of a line

I'll add more tests though

Comment (thomasjo) [reply grammars/tree-sitter-javascript.cson]: Aaaha, thanks for clarifying that :bowtie:

Comment (Aerijo) [reply grammars/tree-sitter-javascript.cson]: Actually, I didn't add tests the first time because none exist yet. I guess I'll (try and) make a new set for Tree-sitter.

Comment (Aerijo) [reply grammars/tree-sitter-javascript.cson]: Emphasis on try. There's no point in testing the tree; that's done by the parser's package. And I'm having trouble getting the scopes.

Comment (maxbrunsfeld) [reply grammars/tree-sitter-javascript.cson]: I'm ok with not having tests on these scope mappings. The actual *parsing* is tested very exhaustively, but testing these kinda feels like testing CSS.

If you have a strong desire to add tests, I'm ok with it, but I'm also cool merging this as-is.

Comment (Aerijo) [reply grammars/tree-sitter-javascript.cson]: - I think tests are a good idea for the Atom specific behaviour, but was [unable to get it working](https://github.com/atom/atom/issues/18443)
- This issue can be raised on almost all the repos though, as they all treat comments as blocks

**Comment by Aerijo** [comment]
OK, I've given up on tests. I still think they're a good idea in the long run, but this change is pretty benign.

@maxbrunsfeld I plan to raise this on all the repos using TS, as there is nothing particularly special about JavaScript.

**Comment by daviwil** [comment]
Looks good to me, thanks @Aerijo!
